log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -238,15 +260,44 @@ private MessageAndSource queuePollFirst()
         }
 
         MessageAndSource message = queue.pollFirst();
```

Comment:
The `queuePollFirst` method should be annotated as `@GuardedBy("queueDrainLock")`